src/main/java/com/bina/lrsim/simulator/Simulator.java

```diff
@@ -45,101 +45,117 @@ public Simulator(RandomFragmentGenerator seqGen) {
    * @param path output path
    * @param movieName movie name
    * @param firsthole first hole producing sequence
-   * @param drawer an instance from which samples can be drawn
+   * @param samplesDrawer an instance from which samples can be drawn
    * @param totalBases minimum number of bp to generate
-   * @param gen random number generator
+   * @param randomNumberGenerator random number generator
    */
-  public int simulate(final String path, final String movieName, final int firsthole, SamplesDrawer drawer, int totalBases, final Spec spec, RandomGenerator gen) throws IOException {
-    try (ReadsWriter writer = ReadsWriterFactory.makeWriter(spec, new File(path, movieName + spec.getSuffix()).getPath(), movieName, firsthole, drawer.getNumRunInfo() > 0 ? drawer.getRunInfo(0) : new RunInfo())) {
+  public int simulate(final String path, final String movieName, final int firsthole,
```

Comment:
one should keep the ```simulate``` public interface intact. Instead, use a simple ```if``` statement to dispatch w.r.t. ```Spec```. In this change, you should make clear that the git-diff is 100% clear that the old code is unchanged. This can be done by 1) renaming this line implementation as ```private int simulateWithPerfectAdapter``` wth zero change otherwise 2) name your new implementation as ```private int simulateWithNoisyAdapter```. 3) somewhere else, implement ```public simulate(...``` as a simple if statement that switch between the two.